query/src/sql/sql_parser.rs

```diff
@@ -236,6 +238,40 @@ impl<'a> DfParser<'a> {
         Ok(DfStatement::Explain(explain_plan))
     }
 
+    // parse show databases where database = xxx or where database
+    fn parse_show_databases(&mut self) -> Result<DfStatement, ParserError> {
+        if self.parser.parse_keyword(Keyword::WHERE) {
+            let mut expr = self.parser.parse_expr()?;
+
+            log::debug!("expr before replace:{}", expr);
+            expr = self.replace_show_database(&expr, "Database", "name")?;
+            log::debug!("expr after replace:{}", expr);
+
+            Ok(DfStatement::ShowDatabases(DfShowDatabases {
+                where_opt: Some(expr),
+            }))
+        } else if self.parser.parse_keyword(Keyword::LIKE) {
+            let pattern = self.parser.next_token().to_string().replace("'", "");
```

Comment:
Why not use `parse_expr`? what happen follow query?
```
SHOW DATABASES LIKE SUBSTRING('1234567890' FROM 3 FOR 3);
```

Reply:
https://dev.mysql.com/doc/refman/8.0/en/show-databases.html

```
SHOW {DATABASES | SCHEMAS}
    [LIKE 'pattern' | WHERE expr]
```
the pattern is a string, maybe I am wrong, and supporting `expr` is good.

Reply:
👍 This is correct in MySQL. We can keep the current status to compatible with MySQL or support expr